### ISSUE TITLE
Blazor CSS Isolation doesn't work when top element in the page is a component

### ISSUE THREAD

**Issue by porkopek** [dotnet/aspnetcore#29409]
### Describe the bug
Blazor CSS isolation `::deep` combinator  **doesn't work  if the top element in a page is a component**, and not an html element.

Example:
This does not get CSS styles with `::deep` in file `Index.razor.css` applied

file `Index.razor`
```
@page "/index"
<Table Datasource="@Datasource"/>
```

but if we change to this
```
@page "/index"
<div>
<Table Datasource="@Datasource"/>
</div>
```
it works.

It seems that RenderTreeBuilder does not apply the generated unique identifier for the css file to the top element (which should be done when `::deep` combinator it's applied) if it's a component, but it does if it is an html element.

### To Reproduce

See this repo https://github.com/porkopek/CSSIsolationIssue
It's the Blazor Server Side .Net 5.0 template that comes with VS 2019
See the changes made [here](https://github.com/porkopek/CSSIsolationIssue/commit/84d1f13a2a209ffd6cb9321a832a1dc92afefe94)

Basically, I've created two pages `'/working'` and `'/notworking'` where I put the `<Button/>` component

### Further technical details
- ASP.NET Core version 5.0.200-preview.20601.7
- Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
- Microsoft Visual Studio Community 2019, Version 16.8.4

**Comment by javiercn** [comment]
@porkopek thanks for contacting us.

CSS isolation doesn't apply to components, only to markup on a given component. In your case, you have a component nested within a component (which might itself be isolated too) and CSS isolation won't cross the boundary between those two.

An HTML element is required to hold the attribute used for CSS isolation since we don't know what the contents of the component are or have no guarantee that the component will apply any attribute we pass as parameter.

Blazor doesn't have any knowledge at runtime about CSS isolation so the scope attribute is just another attribute and the runtime doesn't have any way to enforce a component applies it to the elements it adds to the document.

**Comment by porkopek** [comment]
Thanks for your explanation
